Source/ACE/Entity/Player.cs

```diff
@@ -249,9 +249,23 @@ public Player(Session session) : base(ObjectType.Creature, session.CharacterRequ
         /// <param name="amount">A unsigned long containing the desired XP amount to raise</param>
         public void GrantXp(ulong amount)
         {
-            var test = amount.ToString().Length;
-            character.GrantXp(amount);
-            CheckForLevelup();
+            //until we are max level we must make sure that we send 
+            var chart = DatabaseManager.Charts.GetLevelingXpChart();
+            CharacterLevel maxLevel = chart.Levels.Last();
+            if (character.Level == maxLevel.Level)

```

Comment:
To lock the character's experience bar to the "Infinity!" text and prevent it from going negative, we must not go past 191,226,310,247 in the character's total experience. This part of the function allows for additional experience after max level has been reached.